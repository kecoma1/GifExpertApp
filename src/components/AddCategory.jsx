import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState('Look for a gif brah');

    // Handling the input
    const handleInputChange = e => setInputValue(e.target.value);

    // Preventing the default behavior of the submit
    const handleSubmit = e => {
        e.preventDefault();
    
        if (inputValue.trim().length > 2) {
            setCategories( c => [inputValue, ...c] );
            setInputValue('');
        }
    }

    return (
        <form onSubmit={handleSubmit}>
            <h2>AddCategory</h2>
            <input 
                type="text"
                value={inputValue}
                onChange={ handleInputChange }
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
