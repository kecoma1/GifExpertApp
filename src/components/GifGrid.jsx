import React from 'react';
import PropTypes from 'prop-types';

import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

    const { data, loading } = useFetchGifs( category );

    return (
        <>
            <h3>{ category }</h3>

            { loading && <p>Loading</p> }

            <div className='card-grid'>
                {
                    data.map( img => 
                        <GifGridItem 
                        key={img.id}
                        { ...img }/>
                    )
                }
            </div>
        </>
    );
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}
