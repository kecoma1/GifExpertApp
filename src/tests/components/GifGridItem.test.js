import React from 'react';
import { shallow } from "enzyme";
import { GifGridItem } from '../../components/GifGridItem';


describe('Tests of the component <GifGridItem />:', () => {

    const title = 'titulo';
    const url   = 'https://aaa';
    const wrapper = shallow( <GifGridItem title={title} url={url} /> );

    test('Should show the component correctly', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('Should have an img with the same url and alt of the props', () => {
        const img = wrapper.find('img');
        const { src, alt } = img.props();

        expect( src ).toBe( url );
        expect( alt ).toBe( title );
    });

    test('should have the classes card animate__animated animate__fadeIn', () => { 
        const div = wrapper.find('div');
        const classes = div.prop('className')

        expect( classes.includes('animate__fadeIn') ).toBe( true );
        expect( classes.includes('animate__animated') ).toBe( true );
        expect( classes.includes('card') ).toBe( true );
    });
});