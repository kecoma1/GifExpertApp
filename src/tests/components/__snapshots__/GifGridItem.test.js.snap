// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests of the component <GifGridItem />: Should show the component correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="titulo"
    src="https://aaa"
  />
</div>
`;
